lib/crypto.js

```diff
@@ -652,6 +672,23 @@ function filterDuplicates(names) {
   }).sort();
 }
 
+// timingSafeEqual reuses the timing-safe Hmac.equals function (see above) for
+// comparison of non-hash inputs, like capability URLs or authentication tokens.
+// Note that the strength of the double-HMAC construction does not depend
+// on the unpredictability of the key. Since randomBytes can potentially
+// throw an exception, we use Math.random, which sacrifices nothing in timing
+// safety and benefits from never throwing an exception.
+// See discussion at
+// https://github.com/nodejs/node-v0.x-archive/issues/8560#issuecomment-60145816
+exports.timingSafeEqual = function(a, b) {
+  var key = new Buffer(32);
+  for (var i = 0; i < 32; i += 4) {
+    key.writeFloatBE(Math.random(), i);
```

Comment:
`timingSafeEqual` can still throw, even without `crypto.randomBytes` (which I'm not even sure if that throws regularly). Next-best introduces a possible attack vector through `Math.random`.

That is interesting, about the strength of this. Ultimately it is up to you whether or not you want to change it. Perhaps try a benchmark and see the difference?